### DISCUSSION TITLE
Does a component index file bundle every component if a single one is imported on a page

### DISCUSSION THREAD

**Discussion by JoeRoddy** [vercel/next.js#49899] [Help]
### Summary

We have a component index file for easy importing, like:
`import { Avatar, Button, Logo } from 'components/common';`

The file is like so:
```tsx
import Avatar from './Avatar/Avatar';
import Button from './Button/Button';
// ...

export {
  Avatar,
  Button,
  // ...
}
```

This file is **very** large, we have about 60 components exported from it.  Our app was getting pretty slow and through annoying trial and error, I realized any page that imported a component from the index file, was taking 3+ seconds to load, whereas if it didn't it would be near instant.  If we import the files individually from their explicit paths, it's still instant.

This leads me to believe that if we import one component via our index, every single component is getting bundled.

Is this right? If so, is there any way around this?

**Comment by devjiwonchoi** [comment]
Right, if you export them in one file, the components will be bundled altogether even though some might not be in use.

I would recommend you use an import alias for the components folder.

This [document](https://nextjs.org/docs/app/building-your-application/configuring/absolute-imports-and-module-aliases#module-aliases) is probably the solution you are looking for.

**Comment by icyJoseph** [comment]
This barrel file strategy has been put to question lately, because it makes a huge assumption, which is that, tree shaking will take care of everything, and that's not necessarily true. 

https://steven-lemon182.medium.com/are-typescript-barrel-files-an-anti-pattern-72a713004250

For example, if you run Jest tests, everything will be parsed too.
